### ISSUE TITLE
m3u parser, allow to define custom parsing rules

### ISSUE THREAD

**Issue by FezVrasta** [Fleker/CumulusTV#269]
Right now I have a small `now.sh` app that fetches the URL my IPTV provider gives me, reads the info, and, in my case, moves the group name from the title to the GROUP field.

I think it would be awesome to allow to define a script (LUA? JS? Perl? anything?) that takes the parsed M3U playlist (JSON format) and outputs a modified JSON playlist.

Do you think is this something you would merge in the app? If so I think I would like to try to work on it.

**Comment by Fleker** [comment]
Ideally users shouldn't have to worry about the JSON. It should just be an
internal structure I use.

But I can see value in having user-defined parsing. I'm just not sure the
best way to do it, whether in the app or through a web service on the
website.

**Comment by FezVrasta** [comment]
I would like it to be local (trough a custom script) because it reduces the latency. But even a web service always on would be good I guess.. except for providers that check from where you connect and block you to avoid account sharing probably 🤔

**Comment by Fleker** [comment]
Local support would require some way to execute user scripts on device.

**Comment by FezVrasta** [comment]
Yes that's why I linked to sl4a, it allows to run scripts from Android, I think it is what we need.

**Comment by Fleker** [comment]
If you can take the lead on implementation that would be great.

**Comment by FezVrasta** [comment]
I'll try to work on it this weekend!

Do you have any indications on how to proceed? Preferences? Anything?

**Comment by Fleker** [comment]
I have a couple of test streams that I use to test the parser. I'd
recommend having some unit tests that can verify functionality.

**Comment by FezVrasta** [comment]
Sorry I just tried to run the project but looks like Android Studio can't work if I have Docker installed... I'm very sorry but I'm not able to help 😥
